mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -815,16 +862,26 @@ void mav_vehicle::request_mission_item(uint16_t item_id)
     print_verbose("Mission waypoint requested\n");
 }
 
-void mav_vehicle::arm_throttle()
+void mav_vehicle::arm_throttle(bool arm_disarm)
 {
     send_cmd_long(MAV_CMD_COMPONENT_ARM_DISARM, float(arm_disarm), 0, 0, 0, 0,
                   0, 0, request_intervals_ms::arm_disarm);
 }
 
 void mav_vehicle::takeoff()
```

Comment:
I still need to take a closer look into this to make sure it will work  succesfully with PX4 and Ardupilot.